lib/pf/provisioner/mobileconfig.pm

```diff
@@ -245,6 +265,25 @@ sub _build_profile_template {
     return "wireless-profile-noeap.xml";
 }
 
+sub generate_ipsk {
+    my ($self,$username) = @_;
+    my $person = person_view($username);
+    if (defined $person->{psk} && $person->{ipsk} ne '') {
+        return $person->{psk};
+    }
+    else {
+        my $psk_size;
+        if ($self->psk_size >= 8) {
+            $psk_size = $self->psk_size;
+        } else {
+            $psk_size = 8;
+        }
+        my $psk = word(8,$psk_size);
+        person_modify($username,psk => $psk);
+        return $psk;
```

Comment:
An info logging message saying we've generated key K for user Y would be helpful

Not to confuse with generating KY